thirdparty/runtime_environments.go

```diff
@@ -127,19 +131,35 @@ func (c *RuntimeEnvironmentsClient) getPackages(ctx context.Context, opts Packag
 	return packages, nil
 }
 
+// OSInfoFilterOptions represents the filtering options for GetOSInfo. Each argument is optional except for the AMI field.
+type OSInfoFilterOptions struct {
+	AMI   string
+	Name  string
+	Page  int
+	Limit int
+}
+
 // OSInfo stores operating system information.
 type OSInfo struct {
-	Version string
-	Name    string
+	Version string `json:"version"`
+	Name    string `json:"name"`
 }
 
-// getOSInfo returns a list of operating system information for an AMI.
-func (c *RuntimeEnvironmentsClient) getOSInfo(ctx context.Context, amiID string, page, limit int) ([]OSInfo, error) {
+// GetOSInfo returns a list of operating system information for an AMI.
+func (c *RuntimeEnvironmentsClient) GetOSInfo(ctx context.Context, opts OSInfoFilterOptions) ([]OSInfo, error) {
+	if opts.AMI == "" {
+		return nil, errors.New("no AMI provided")
+	}
 	params := url.Values{}
-	params.Set("ami", amiID)
-	params.Set("page", strconv.Itoa(page))
-	params.Set("limit", strconv.Itoa(limit))
+	params.Set("ami", opts.AMI)
+	if opts.Page != 0 {
+		params.Set("page", strconv.Itoa(opts.Page))
+	}
```

Comment:
Should we be adding this condition to the other functions as well ?

Reply:
Realized I don't need this check because opts.Page defaults to 0.